### ISSUE TITLE
no-initialize-state - false positive on async code

### ISSUE THREAD

**Issue by nirtamir2** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#23]
Hi
I have this code 
```tsx
  const [userback, setUserback] = useState<UserbackWidget | null>(null);

  useEffect(() => {
    async function initUserback() {
      const instance = await Userback(env.NEXT_PUBLIC_USERBACK_KEY);
      setUserback(instance);
    }

    void initUserback();
  }, []);
```

I need to wait for the Userback promise to resolve to initialize the state. 

This component is client only init script

Is it false positive - or what is the alternative?

**Comment by NickvanDyke** [comment]
I think you're correct. IME libraries usually provide better higher-level ways to initialize and pass around their state, but in this case, this may be your only option.

**Comment by nirtamir2** [comment]
Yes, I will do this (I started it at first).
I thought `const userback = use(Userback(env.NEXT_PUBLIC_USERBACK_KEY))` would work too but it didn't.
Thanks!

> Handling async code this way is often riddled with bugs and subtle race condition-related issues. I would strongly advise the usage of react-query here, i.e.
> 
> import { useSuspenseQuery } from '@tanstack/react-query';
> 
> ...
> 
> const { data: userback } = useSuspenseQuery({
>   queryKey: ['userback'],
>   queryFn: async () => await Userback(env.NEXT_PUBLIC_USERBACK_KEY),
> });
> 
> ...
> or as [@NickvanDyke](https://github.com/NickvanDyke) suggested, this may work fine in your case too (though top-level await / es2022 is required):
> 
> const userback = await Userback(env.NEXT_PUBLIC_USERBACK_KEY);
> 
> function MyComponent() {
>   ...
> }
